### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by auscompgeek**
Huge refactor of the code so things actually make sense.

Stop adding a highly inefficient global listener on NetworkTables.

Fixes the auto-checklisting to only auto-check in the correct state.

Also removes all CSS from the JavaScript, where it doesn't belong.

Also move some things around for driver convenience hopefully.

### CODE REVIEW

File: js/dashboard.js [hunk 1]
```diff
@@ -12,36 +7,41 @@ sports_music.setAttribute('src', 'music/Sports.ogg');
...
-        } else {
+        if (!connected) {
             $("#check-connection").prop("checked", false);
         }
     }, true);
-
-    autoChecker();
+    NetworkTables.addKeyListener(
+        "/robot/is_ds_attached",
+        (key, value) => $("#check-connection").prop("checked", value),
```
[the remark this answers is not in the document]
Comment (auscompgeek) [reply js/dashboard.js:43]: pls listen to correct jshintrc

File: js/dashboard.js [hunk 2]
```diff
@@ -12,36 +7,41 @@ sports_music.setAttribute('src', 'music/Sports.ogg');
...
 
-    $("input").click(function () {
-        if ($("input").length === $("input:checked").length) {
-            removeForm();
+    $("#checklist input").click(function () {
+        if ($("#checklist input").length === $("#checklist input:checked").length) {
+            hideChecklist();
         }
     });
+    NetworkTables.addKeyListener("/SmartDashboard/Autonomous Mode/selected", () => $("#check-auto").prop("checked", true));
```
[the remark this answers is not in the document]
Comment (fizzlebert) [reply js/dashboard.js:34]: How come the autonomous selector did not check the box when I changed the auto mode.

[the remark this answers is not in the document]
Comment (auscompgeek) [reply js/dashboard.js:34]: Did you update your pynetworktables2js installation?

File: index.html [hunk 3]
```diff
@@ -11,23 +11,21 @@
...
 
-    <div id="camera" class="camera inital-hide"></div>
-
-    <form class="checklist-div" id="checklist">
+    <form id="checklist">
         <label><input id="check-connection" type="checkbox"> Is robot connected?<br></label>
-        <label><input id="auto" type="checkbox"> Select Auto<br></label>
+        <label><input id="check-auto" type="checkbox"> Select Auto<br></label>
         <label><input type="checkbox"> Is cube loaded?<br></label>
         <label><input type="checkbox"> Is controller connected and in correct order.?<br></label>
```
Comment (fizzlebert) [inline index.html:30]: Also this should be changed to Is controller connected and in correct order?

Comment (auscompgeek) [reply index.html:30]: Well I never touched now, did I.

**Pull request merged**